libs/geospatial/src/main/java/org/codice/ddf/libs/location/CountryCodeConverter.java

```diff
@@ -0,0 +1,105 @@
+
+    private static Map<String, String> countryCodesMap;
+
+    /**
+     * Converts FIPS 10-4 into ISO 3166-1 alpha 3.
+     *
+     * @param fipsCountryCode a FIPS 10-4 country code
+     * @return an ISO 3166-1 alpha-3 country code
+     */
+    public static String convertFipsToIso3(String fipsCountryCode) {
+        String iso3CountryCode = getCountryCodesMap().get(fipsCountryCode);
+        if (StringUtils.isBlank(iso3CountryCode)) {
+            throw new MissingResourceException(
+                    "Could not find a corresponding ISO 3166-1 alpha 3 country code given the FIPS"
+                            + " 10-4 country code: " + fipsCountryCode,
+                    CountryCodeConverter.class.toString(),
+                    fipsCountryCode);
+        }
+        return iso3CountryCode;
+    }
+
+    /**
+     * Converts ISO 3166-1 alpha 3 into FIPS 10-4.
+     * <p>
+     * Special case:
+     * ISO 3166-1 alpha 3 country: "PSE" corresponds to FIPS 10-4 countries: "WE" and "GZ",
+     * in this case, the returned list will contain those two countries. In all other cases,
+     * the returned list will contain one country.
+     *
+     * @param iso3alphaCountryCode a FIPS 10-4 country code
+     * @return an ISO 3166-1 country code in a List<String>
+     */
+    public static List<String> convertIso3ToFips(String iso3alphaCountryCode) {
+        if ("PSE".equals(iso3alphaCountryCode)) {
+            return Arrays.asList("WE", "GZ");
+        }
+        String fipsCountryCode = getCountryCodesMap().entrySet()
+                .stream()
+                .filter(entry -> Objects.equals(entry.getValue(), iso3alphaCountryCode))
+                .map(Map.Entry::getKey)
+                .findFirst()
+                .orElse(null);
+        if (StringUtils.isBlank(fipsCountryCode)) {
+            throw new MissingResourceException(
+                    "Could not find a corresponding FIPS 10-4 country code given the ISO 3166-1"
+                            + " alpha 3 country code: " + iso3alphaCountryCode,
+                    CountryCodeConverter.class.toString(),
+                    iso3alphaCountryCode);
+        }
+        return Arrays.asList(fipsCountryCode);
+    }
+
+    /**
+     * Converts ISO 3166-1 alpha 2 into ISO 3166-1 alpha 3.
+     *
+     * @param language          the language to return the converted country code in
+     * @param alpha2CountryCode ISO 3166-1 alpha-2 formatted country code
+     * @return an ISO 3166-1 alpha-3 country code
+     */
+    public static String convertIso2ToIso3(String language, String alpha2CountryCode) {
+        Locale locale = new Locale(language, alpha2CountryCode);
+        return locale.getISO3Country();
+    }
+
+    private static Map<String, String> getCountryCodesMap() {
+        if (countryCodesMap == null || countryCodesMap.isEmpty()) {
+            countryCodesMap = PropertiesLoader.toMap(PropertiesLoader.loadProperties(FIPS_TO_ISO3));
```

Comment:
My current solution will load the countryCodesMap once, because once the static variable has been initialized it will never be null again. However, this lazy init is not thread safe without it being synchronized, so I will put it in a static block.